gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StatisticsTrackingHttpRequestInitializer.java

```diff
@@ -15,50 +15,37 @@
  */
 package com.google.cloud.hadoop.gcsio;
 
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
 import com.google.api.client.http.HttpRequest;
 import com.google.api.client.http.HttpRequestInitializer;
-import com.google.api.client.http.HttpResponse;
 import com.google.common.base.Ascii;
+import com.google.common.collect.ImmutableMap;
 import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.atomic.AtomicLong;
 
-/** To track and update the statistics related to http requests */
+/** Tracks and updates the statistics related to HTTP requests */
 class StatisticsTrackingHttpRequestInitializer implements HttpRequestInitializer {
 
-  private final ConcurrentMap<String, AtomicLong> statistics;
-
-  public StatisticsTrackingHttpRequestInitializer(ConcurrentMap<String, AtomicLong> statistics) {
-    this.statistics = statistics;
-  }
+  private final ConcurrentMap<String, AtomicLong> statistics = new ConcurrentHashMap<>();
 
   @Override
   public void initialize(HttpRequest request) throws IOException {
     request.setResponseInterceptor(
         httpResponse -> {
-          setHttpRequestStats(httpResponse.getRequest());
-          setHttpRequestFailureStats(httpResponse);
+          String requestMethod = Ascii.toLowerCase(httpResponse.getRequest().getRequestMethod());
+          int statusCode = httpResponse.getStatusCode();
+          statistics
+              .computeIfAbsent("http_" + requestMethod + "_" + statusCode, k -> new AtomicLong())
```

Comment:
So we are removing total request count?

Do we want to keep the existing counters and add this one?

Reply:
I think that previous logic was quite confusing, because request were double-counted in failure and success counters, right now each request counted only once and it's clear what response came for it.